servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -160,9 +173,13 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
       dependency.resetUncheckpointedChildrenFiles(dependencyEntry.mUncheckpointedFiles);
       mDependencyMap.addDependency(dependency);
     } else if (entry instanceof CompleteFileEntry) {
-      completeFileFromEntry((CompleteFileEntry) entry);
-    } else if (entry instanceof AddCheckpointEntry) {
-      completeFileCheckpointFromEntry((AddCheckpointEntry) entry);
+      try {
+        completeFileFromEntry((CompleteFileEntry) entry);
+      } catch (InvalidPathException e) {
+        throw new IOException("failed to complete file " + ((CompleteFileEntry) entry).getFileId());
```

Comment:
Why is this an `IOException` but above failures are `RuntimeExceptions`? Perhaps we should have some kind of `JournalFailureException`?